lib/transloadit/assembly.rb

```diff
@@ -66,6 +68,65 @@ def submit!(*ios)
   end
 
   #
+  # alias for create!
+  # keeping this method for backward compatibility
+  #
+  def submit!(*ios)
+    warn "#{caller(1)[0]}: warning: Transloadit::Assembly#submit!"\
+      " is obsolete use Transloadit::Assembly#create! instead"
+
+    self.create!(*ios)
+  end
+
+  #
+  # Returns a list of all assemblies
+  # @param [Hash]        additional GET data to submit with the request
+  #
+  def list(params = {})
+    params = self.to_hash.update(params)
+    Transloadit::Request.new('/assemblies', self.transloadit.secret).get(params)
+  end
+
+  #
+  # Returns a single assembly object specified by the assembly id
+  # @param [String]     id    id of the desired assembly
+  #
+  def get(id)
+    request = Transloadit::Request.new("/assemblies/#{id}", self.transloadit.secret)
+    request.get.extend!(Transloadit::Response::Assembly)
+  end
+
+  #
+  # Replays an assambly specified by the  id
+  # @param [String]   id       id of the desired assembly
+  # @param [Hash]     params   additional POST data to submit with the request
+  #
+  def replay(id, params = {})
+    params = { :params => { :wait => false }.merge(self.to_hash.update(params)) }
+    request = Transloadit::Request.new("/assemblies/#{id}/replay", self.transloadit.secret)
+    request.post(params).extend!(Transloadit::Response::Assembly)
+  end
+
+  #
+  # Returns all assembly notifications
+  # @param [Hash]        params    additional GET data to submit with the request
+  #
+  def notifications(params = {})
+    params = self.to_hash.update(params)
+    Transloadit::Request.new("/assembly_notifications", self.transloadit.secret).get(params)
```

Comment:
The pattern of

```
params = self.to_hash.update(params)
Transloadit::Request.new(SOME_ENDPOINT, self.transloadit.secret).get(params)
```

is repetitive. Let's encapsulate that in a different function and let these functions call that with their custom endpoint string and the supplied params.